hbase-agent/src/main/java/org/apache/ranger/authorization/hbase/RangerAuthorizationCoprocessor.java

```diff
@@ -90,19 +91,28 @@ public class RangerAuthorizationCoprocessor implements AccessControlService.Inte
 	private static final String GROUP_PREFIX = "@";
 
 	private UserProvider userProvider;
-    private RegionCoprocessorEnvironment regionEnv;
+	private RegionCoprocessorEnvironment regionEnv;
 	private Map<InternalScanner, String> scannerOwners = new MapMaker().weakKeys().makeMap();
 	/** if we should check EXEC permissions */
 	private boolean shouldCheckExecPermission;
-	
+
 	/*
 	 * These are package level only for testability and aren't meant to be exposed outside via getters/setters or made available to derived classes.
 	 */
 	final HbaseFactory _factory = HbaseFactory.getInstance();
 	final HbaseUserUtils _userUtils = _factory.getUserUtils();
 	final HbaseAuthUtils _authUtils = _factory.getAuthUtils();
 	private static volatile RangerHBasePlugin hbasePlugin = null;
-	
+
+	public void setColumnAuthOptimizationEnabled(boolean enable) throws Exception {
+		if (hbasePlugin!=null) {
```

Comment:
Assign RangerAuthorizationCoprocessor. hbasePlugin to a local variable before accessing it:

```
RangerHBasePlugin plugn = RangerAuthorizationCoprocessor.hbasePlugin;

if (plugin != null) {
  plugin. setColumnAuthOptimizationEnabled(enable);
}
```

Reply:
Why is this assigning hbasePlugin to a local variable required ?
hbasePlugin is already initialized in the start()

Reply:
hbasePlugin is a static volatile member, whose value could change between if at line 108 and referencing at line 109. Hence copying to a local variable is a safe practice.

Reply:
Addressed in the latest commit